devito/ops/compiler.py

```diff
@@ -0,0 +1,74 @@
+from codepy.jit import compile_from_string
+from time import time
+import warnings
+
+from devito.logger import debug
+from devito.parameters import configuration
+
+
+__all__ = ['CompilerOPS']
+
+
+class CompilerOPS(configuration['compiler'].__class__):
+    def __init__(self, *args, **kwargs):
+        super(CompilerOPS, self).__init__(*args, **kwargs)
+
+    def jit_compile_ops(self, soname, code, hcode):
+        target = str(self.get_jit_dir().joinpath(soname))
+        src_file = "%s.%s" % (target, self.src_ext)
+        src_file2 = "%s.h" % (target)
+        cache_dir = self.get_codepy_dir().joinpath(soname[:7])
```

Comment:
this entire block of code is copy-paste. Not exploiting inheritance -> mistake

this needs fixing

Reply:
My initial idea was to override the function, but as i utilize an extra parameter i opted for copy-pasting it. Is there a way to overwrite it?